### ISSUE TITLE
How to replace NCCL2 and CUDA with another accelerator driver in Horovod ?

### ISSUE THREAD

**Issue by liuyongheng123** [horovod/horovod#1322]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
I would like to use another set of AI accelerators instead of Nvidia GPUs. Now I can compile Horovod WITHOUT NCCL2 and CUDA. But I still don't know how to add another AI hardware driver library to Horovod so that I can take advantage of ring-allreduce algorithm.

Also I would like to know specificly what do I need to do to design the APIs for Horovod to use my AI accelerators. Many thanks !

**Comment by tgaddair** [comment]
Hey @liuyongheng123, please take a look at our Contributors Guide, specifically the section on [Adding Custom Operations](https://github.com/horovod/horovod/blob/master/docs/contributors.rst#adding-custom-operations).  It explains how to add a new operation for a given framework, and you can take a look at the implementations for MLSL, DDL, and Gloo for guidance.

**Comment by liuyongheng123** [comment]
Thanks a lot!!!
